dask/bytes/tests/test_hdfs.py

```diff
@@ -21,59 +33,108 @@
 basedir = '/tmp/test-dask'
 
 
-@pytest.fixture
-def hdfs():
-    hdfs = hdfs3.HDFileSystem(host='localhost', port=8020)
-    if hdfs.exists(basedir):
-        hdfs.rm(basedir)
-    hdfs.mkdir(basedir)
+HDFSDriver = namedtuple('HDFSDriver', ['hdfs', 'name'])
+
+
+def make_fixture(params):
+    @pytest.fixture(params=params)
+    def _(request):
+        if request.param == 'hdfs3':
+            hdfs = hdfs3.HDFileSystem(host='localhost', port=8020)
+        else:
+            hdfs = pyarrow.hdfs.connect(host='localhost', port=8020)
+
+        if hdfs.exists(basedir):
+            hdfs.rm(basedir, recursive=True)
+        hdfs.mkdir(basedir)
+
+        yield HDFSDriver(hdfs, request.param)
```

Comment:
Maybe instead do the following?

```python
with dask.set_options(hdfs_driver=request.param):
    yield hdfs
```

This would centralize the `dask.set_options` code and remove the need for the `HDFSDriver` class.

(I may be missing something though)

Reply:
Generally agree that it would be nice to have this as accurately match usage as possible.
Would it work to replace the `driver.fs.open` calls below with `open_files`? Then each test can run in one context.

Reply:
I went with Matt's approach, this seemed cleaner and we already test that the `set_options` bit results in the correct driver.